lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp

```diff
@@ -653,8 +703,14 @@ bool SwiftLanguageRuntimeImpl::AddModuleToReflectionContext(
     m_reflection_ctx->readELF(
         swift::remote::RemoteAddress(load_ptr),
         llvm::Optional<llvm::sys::MemoryBlock>(file_buffer));
+  } else if (obj_file->GetPluginName().GetStringRef().equals("mach-o")) {
```

Comment:
is there a way to check this without doing a string comparison?

Reply:
I don't think so, unfortunately... The subclasses of ObjectFile are implemented as Plugins, so SwiftLanguageRuntime doesn't know of their existence. The only way I found to get what implementation it belonged to was through this `GetPluginName` function.